wandb/sdk/interface/artifacts.py

```diff
@@ -908,17 +908,15 @@ def cleanup(self, target_size: int) -> int:
         total_size: int = 0
         for root, _, files in os.walk(self._cache_dir):
             for file in files:
-                path = os.path.join(root, file)
-                stat = os.stat(path)
+                try:
+                    path = os.path.join(root, file)
+                    stat = os.stat(path)
 
-                if file.startswith(ArtifactsCache._TMP_PREFIX):
-                    try:
+                    if file.startswith(ArtifactsCache._TMP_PREFIX):
                         os.remove(path)
                         bytes_reclaimed += stat.st_size
-                    except OSError:
-                        pass
+                except OSError:
```

Comment:
is `FileNotFoundError` an OS error?

Reply:
Yes, as is PermissionError and IsADirectoryError, though I could conservatively specify them individually. I think only those first two are possible.